src/pages/Admin/Tools/Tools.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import firebase from "firebase/app";
+
+import * as scripts from "./scripts";
+import { SelfServeScript } from "./types";
+
+export const Tools: React.FC = () => {
+  const [chosenScript, setChosenScript] = useState<SelfServeScript>();
+
+  return (
+    <div>
+      {chosenScript ? (
+        <Script script={chosenScript} />
+      ) : (
+        Object.values(scripts).map((script) => (
+          <button
+            onClick={() => setChosenScript(script)}
+            key={`${script.name}-${script.functionLocation}`}
+          >
+            {script.name}
+          </button>
+        ))
+      )}
+    </div>
+  );
+};
+
+export const Script: React.FC<{ script: SelfServeScript }> = ({ script }) => {
+  const { register, handleSubmit } = useForm();
+
+  const [links, setLinks] = useState<{ [key: string]: string }>();
+
+  const onSubmit = async (data: Object) => {
```

Comment:
can benefit from `useAsyncFn`

Reply:
Done in https://github.com/sparkletown/sparkle/pull/2747/commits/73bc0106ad8f80a3da7b69b082166b60d78fc4e8